lib/hnvm/config.sh

```diff
@@ -123,22 +149,68 @@ fi
 function resolve_ver() {
   local name=${1}
   local ver=${2}
-  local cache_file="${script_dir}/../.tmp/${name}/${ver}"
-
-  if [[ ! "$ver" =~ [0-9]+\.[0-9]+\.[0-9]+ ]]; then
+  local initial_ver=${2}
+  local ver_sanitized
+  local cache_file
+  ver_sanitized="${ver//[^a-z0-9_]/_}"
+  cache_file="${HNVM_PATH}/.tmp/${name}/${ver_sanitized}"
+
+  # Only spend time resolving the version if it's not a fully resolved version number.
+  if is_invalid_version "$ver"; then
     mkdir -p "$(dirname "${cache_file}")"
 
-    # Cache result
+    # Check if the previous cache result exists and has been modified within the past HNVM_RANGE_CACHE seconds.
     if [ -f "${cache_file}" ] && [ "$(( $(date +"%s") - HNVM_RANGE_CACHE ))" -le "$(date -r "${cache_file}" +"%s")" ]; then
       ver="$(cat "${cache_file}")"
     else
-      echo -e $'\e[33mWarning\e[0m: Resolving '"${name}"' version range "'"${ver}"'" is slower than providing an exact version.' > ${COMMAND_OUTPUT}
+      echo -e $'\e[33mWarning\e[0m: Resolving '"${name}"' "'"${ver}"'" is slower than providing an exact version.' > ${COMMAND_OUTPUT}
```

Comment:
Hmm, this is a lot of green. I see why you wanted to add the tests. Can we split this up into two PR's? One that adds the tests without changing much implementation, and then one that changes things up? That'll make me more confident we aren't unknowingly changing something here.

Reply:
this actually is two PRs already 😃 (see #23), just the tests won't pass until this is fixed so I wanted to try the PR with everything working first. 

I'll disable the broken tests and remove this commit before landing then re-enable with #23 👍